### PR TITLE
(PXP-7952): feat/downloadCallIndexd: have File download button call indexd

### PR DESCRIPTION
**Pull request by ocshawn**
Jira Ticket: [PXP-7952](https://ctds-planx.atlassian.net/browse/PXP-7952)
### Improvement
- have file download button call indexd to check if user can download file

### CODE REVIEW

File: src/CoreMetadata/CoreMetadataHeader.jsx [hunk 1]
```diff
@@ -40,36 +48,48 @@ class CoreMetadataHeader extends Component {
...
-          );
-        }
-      }
+
+      // check if user has permision to download file
+      fetch(`/index/index/${this.props.metadata.object_id}`)
+        .then((response) => response.json())
+        .then((data) => {
+          if (data.authz && data.authz.length > 0) {
+            const userHasAccess = data.authz.every((urlTocheck) => userHasMethodForServiceOnResource('read-storage', 'fence', urlTocheck, this.props.userAuthMapping));
```
Comment (paulineribeyre) [inline src/CoreMetadata/CoreMetadataHeader.jsx:59]: `urlTocheck` => i would rename to `resource` for accuracy

Comment (paulineribeyre) [reply src/CoreMetadata/CoreMetadataHeader.jsx:59]: otherwise lgtm

File: src/CoreMetadata/CoreMetadataHeader.jsx [hunk 2]
```diff
@@ -40,36 +48,48 @@ class CoreMetadataHeader extends Component {
...
-              label={SIGNED_URL_BTN_CAPTION}
-              className='core-metadata-page__column--right--signed-url-button'
-              buttonType='primary'
-            />
-          );
-        }
-      }
+
+      // check if user has permision to download file
+      fetch(`/index/index/${this.props.metadata.object_id}`)
```
Comment (mfshao) [inline src/CoreMetadata/CoreMetadataHeader.jsx:55]: you should use `indexdPath` from localconf for this

**Review by mfshao — COMMENTED** [review]
Did we know if we have completelyu move away from `acl` to `authz` for indexd records everywhere?

Portal has a config `useIndexdAuthz` to determine whether to user `acl` or `authz` for indexd records for certain operations like submission, and it was default to use `acl` because some of indexd records don't have `authz` at that time

**Comment by mfshao** [comment]
and nitpicking: this is more like an `Improvement` rather than `New feature`

File: src/CoreMetadata/CoreMetadataHeader.jsx [hunk 3]
```diff
@@ -44,52 +44,69 @@ class CoreMetadataHeader extends Component {
...
+    const { projectAvail } = this.props;
+    const projectId = this.props.metadata.project_id;
+    fetch(`${indexdPath}${this.props.metadata.object_id}`)
+      .then((response) => response.json())
+      .then((data) => {
+        let userHasAccess = false;
+        if (data.authz && data.authz.length > 0) {
+          userHasAccess = data.authz.every((resource) => userHasMethodForServiceOnResource('read-storage', 'fence', resource, this.props.userAuthMapping));
+        } else {
+          // if no authz fall back to old revivification method
```
Comment (paulineribeyre) [inline src/CoreMetadata/CoreMetadataHeader.jsx:62]: revivification?

**Review by paulineribeyre — COMMENTED** [review]
lgtm but i'll let @mfshao approve

**Review by mfshao — APPROVED** [review]

**Pull request merged**